src/GoTrueClient.ts

```diff
@@ -632,7 +644,7 @@ export default class GoTrueClient {
         } else {
           this._removeSession()
         }
-      } else if (!currentSession || !currentSession.user) {
```

Comment:
@kiwicopple do you recall why this check was introduced? Is it fine to be removed? Currently the verify methods to not return the user object. While @kangmingtay is making a PR in gotrue to return the user object on the verify endpoints (https://github.com/supabase/gotrue/issues/141) it likely won't be available to users until next week. So we're contemplating removing this check here so this issue is resolved before the gotrue server changes go live.